src/transaction/log_page_buffer.c

```diff
@@ -3361,6 +3364,250 @@ logpb_write_toflush_pages_to_archive (THREAD_ENTRY * thread_p)
     }
 }
 
+#if defined(SERVER_MODE)
+/*
+ * logpb_catchup_append_page - Append a page during catch-up.
+ *
+ *   pgptr(in): a log page pointer to append
+ */
+void
+logpb_catchup_append_page (THREAD_ENTRY * thread_p, const LOG_PAGE * const log_pgptr)
+{
+  assert (LOG_CS_OWN_WRITE_MODE (thread_p));
+  assert (log_Gl.append.log_pgptr != NULL);
+
+  logpb_catchup_start_append (thread_p);
+
+  if (log_Gl.append.log_pgptr->hdr.logical_pageid + 1 == log_pgptr->hdr.logical_pageid)
+    {
+      // The normal case. Only exception is the first page. 
+      log_Gl.append.prev_lsa = log_Gl.hdr.append_lsa;
+      logpb_next_append_page (thread_p, LOG_SET_DIRTY);
+    }
+  else if (log_Gl.append.log_pgptr->hdr.logical_pageid == log_pgptr->hdr.logical_pageid)
+    {
+      // For the first page, just overwrite the page.
+      // DON'T change meta data like log_Gl.append.prev_lsa. They don't go backward
+    }
+  else
+    {
+      logpb_fatal_error (thread_p, true, ARG_FILE_LINE, "logpb_catchup_append_page");
```

Comment:
I understood the other if/else scopes.
My comment was specifically about this `logpb_fatal_error`.
The way it crashes the application is both drastic and laconic - with just the `logpb_catchup_append_page` message which does not say much.

But also my "Could you add some explanation for this branch?" was cryptic because I meant to ask "Could you add some __comment__ for this branch?":
- commenting why the fatal error
- maybe changing te logged message to something more meaningful _ "... not yet implemented", "not possible to occur" ...
- or a TODO for the future (if the case)

Reply:
Oh, I see.